src/main/kotlin/com/greenbird/utilihive/stubs/sftp/SftpServerTestContext.kt

```diff
@@ -0,0 +1,323 @@
+        path: String,
+        content: ByteArray
+    ) {
+        verifyWithSftpServerIsNotFinished("upload file")
+        val pathAsObject = fileSystem.getPath(path)
+        ensureDirectoryOfPathExists(pathAsObject)
+        Files.write(pathAsObject, content)
+    }
+
+    /**
+     * Puts a file to the SFTP folder. The file is available by the specified
+     * path. The file's content is read from an `InputStream`.
+     * @param path the path to the file
+     * @param inputStream an `InputStream` that provides the file's content
+     * @throws IOException if the file cannot be written or the input stream cannot be read
+     */
+    @Throws(IOException::class)
+    fun putFile(
+        path: String,
+        inputStream: InputStream
+    ) {
+        verifyWithSftpServerIsNotFinished("upload file")
+        val pathAsObject = fileSystem.getPath(path)
+        ensureDirectoryOfPathExists(pathAsObject)
+        Files.copy(inputStream, pathAsObject)
+    }
+
+    /**
+     * Creates a directory on the SFTP server.
+     * @param path the directory's path
+     * @throws IOException if the directory cannot be created
+     */
+    @Throws(IOException::class)
+    fun createDirectory(
+        path: String
+    ) {
+        verifyWithSftpServerIsNotFinished("create directory")
+        val pathAsObject = fileSystem.getPath(path)
+        Files.createDirectories(pathAsObject)
+    }
+
+    /**
+     * Create multiple directories on the SFTP server.
+     * @param paths the directories' paths.
+     * @throws IOException if at least one directory cannot be created.
+     */
+    @Throws(IOException::class)
+    fun createDirectories(
+        vararg paths: String
+    ) {
+        for (path in paths)
+            createDirectory(path)
+    }
+
+    /**
+     * Gets a text file's content from the SFTP server. The content is decoded
+     * using the specified encoding.
+     * @param path the path to the file
+     * @param encoding the file's encoding
+     * @return the content of the text file
+     * @throws IOException if the file cannot be read
+     */
+    @Throws(IOException::class)
+    fun getFileContent(
+        path: String,
+        encoding: Charset
+    ): String = getFileContent(path).toString(encoding)
+
+    /**
+     * Gets a file from the SFTP server.
+     * @param path the path to the file
+     * @return the content of the file
+     * @throws IOException if the file cannot be read
+     */
+    @Throws(IOException::class)
+    fun getFileContent(
+        path: String
+    ): ByteArray {
+        verifyWithSftpServerIsNotFinished("download file")
+        val pathAsObject = fileSystem.getPath(path)
+        return Files.readAllBytes(pathAsObject)
+    }
+
+    /**
+     * Checks the existence of a file. Returns `true` iff the file exists
+     * and it is not a directory.
+     * @param path the path to the file
+     * @return `true` iff the file exists and it is not a directory
+     */
+    fun existsFile(
```

Comment:
Change name to `exists` or something?  Alternatively, and probably better,  we can provide a function that returns a `Path` and then the users can do checks like this (and many others) on that?

Reply:
Good idea, added a function returning Path.